changelog/_unreleased/2025-11-03-sw-currency-header-always-overrides-context.md

```diff
@@ -0,0 +1,7 @@
+---
+title: sw-currency-id header always overrides context values
+---
+# Core
+* Added `\Shopware\Core\System\SalesChannel\Context\SalesChannelContextServiceParameters::__construct(overwriteCurrencyId)` parameter to force overwrite of the currency id.
+* Changed `\Shopware\Core\Framework\Routing\SalesChannelRequestContextResolver::resolve()` to always set the overwrite currency id from the request header.
+* Changed `\Shopware\Core\System\SalesChannel\Context\SalesChannelContextService::get()` to always overwrite the currency id if the overwrite currency parameter is set.
```

Comment:
definitely worth a `Upgrade information` section, or the new files, if already used